### PR TITLE
[TGL] Update Vtd support

### PR DESCRIPTION
**Pull request by gdong1**
GNVS should be aligned with VTD PCD
Fix a DMR check issue.

Signed-off-by: Guo Dong <guo.dong@intel.com>

### CODE REVIEW

File: Silicon/TigerlakePkg/Library/Stage2SocInitLib/Stage2SocInitLib.c [hunk 1]
```diff
@@ -70,7 +70,7 @@ SocUpdateAcpiGnvs (
     McD0BaseAddress  = PCI_LIB_ADDRESS (SA_MC_BUS, 0, 0, 0);
     McD2BaseAddress  = PCI_LIB_ADDRESS (SA_IGD_BUS, SA_IGD_DEV, SA_IGD_FUN_0, 0);
 
-    Gnvs->SaNvs.VtdDisable = FALSE;
+    Gnvs->SaNvs.VtdDisable = !((FeaturePcdGet (PcdVtdEnabled)));
```
Comment (mauricema) [inline Silicon/TigerlakePkg/Library/Stage2SocInitLib/Stage2SocInitLib.c:73]: Remove redundant bracket ?

File: Silicon/TigerlakePkg/Library/VTdLib/VTdLib.c [hunk 2]
```diff
@@ -117,7 +117,7 @@ UpdateDrhd (
     NeedRemove = TRUE;
   }
   if ((DrhdEngine->DeviceScope[0].PciPath.Device == SA_IGD_DEV) && (DrhdEngine->DeviceScope[0].PciPath.Function == SA_IGD_FUN_0) &&
-      (PciRead32 (PCI_LIB_ADDRESS (0, DrhdEngine->DeviceScope[0].PciPath.Device, DrhdEngine->DeviceScope[0].PciPath.Function, 0x00)) == 0xFFFFFFFF)) {
+      (PciRead32 (PCI_LIB_ADDRESS (0, DrhdEngine->DeviceScope[0].PciPath.Device, DrhdEngine->DeviceScope[0].PciPath.Function, 0x00)) != 0xFFFFFFFF)) {
```
Comment (mauricema) [inline Silicon/TigerlakePkg/Library/VTdLib/VTdLib.c:120]: This line is too long.  Consider using a variable VidDid instead.

File: Silicon/TigerlakePkg/Library/Stage2SocInitLib/Stage2SocInitLib.c (hunk 1 above)
Comment (gdong1) [reply Silicon/TigerlakePkg/Library/Stage2SocInitLib/Stage2SocInitLib.c:73]: Sure. will update

File: Silicon/TigerlakePkg/Library/VTdLib/VTdLib.c (hunk 2 above)
Comment (gdong1) [reply Silicon/TigerlakePkg/Library/VTdLib/VTdLib.c:120]: sure. will update

**Review by myguitar — APPROVED** [review]
Looks good to me

**Pull request merged**